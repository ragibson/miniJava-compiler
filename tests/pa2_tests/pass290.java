// PA2 pass AST reflects operator precedence
class A {
    public static void main(String [] args) {
        // these two stmts should have the same expr AST
        boolean b = false || true == 2 < -3 - 4 / 5 && !!false;
        boolean b = false || ((true == (2 < ((-3) - (4 /5)))) && (!(!false)));
    }
}
