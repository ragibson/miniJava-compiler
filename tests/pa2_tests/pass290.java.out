********** miniJava Compiler (PA2) **********
Syntactic Analysis ... 
Syntactic Analysis OK
======= AST Display =========================
Package
  ClassDeclList [1]
  . ClassDecl
  .   "A" classname
  .   FieldDeclList [0]
  .   MethodDeclList [1]
  .   . (public static) MethodDecl
  .   .   VOID BaseType
  .   .   "main" methodname
  .   .   ParameterDeclList [1]
  .   .   . ParameterDecl
  .   .   .   ArrayType
  .   .   .     ClassType
  .   .   .       "String" Identifier
  .   .   .   "args"parametername 
  .   .   StmtList [2]
  .   .   . VarDeclStmt
  .   .   .   VarDecl
  .   .   .     BOOLEAN BaseType
  .   .   .     "b" varname
  .   .   .   BinaryExpr
  .   .   .     "||" Operator
  .   .   .       LiteralExpr
  .   .   .         "false" BooleanLiteral
  .   .   .       BinaryExpr
  .   .   .         "&&" Operator
  .   .   .           BinaryExpr
  .   .   .             "==" Operator
  .   .   .               LiteralExpr
  .   .   .                 "true" BooleanLiteral
  .   .   .               BinaryExpr
  .   .   .                 "<" Operator
  .   .   .                   LiteralExpr
  .   .   .                     "2" IntLiteral
  .   .   .                   BinaryExpr
  .   .   .                     "-" Operator
  .   .   .                       UnaryExpr
  .   .   .                         "-" Operator
  .   .   .                           LiteralExpr
  .   .   .                             "3" IntLiteral
  .   .   .                       BinaryExpr
  .   .   .                         "/" Operator
  .   .   .                           LiteralExpr
  .   .   .                             "4" IntLiteral
  .   .   .                           LiteralExpr
  .   .   .                             "5" IntLiteral
  .   .   .           UnaryExpr
  .   .   .             "!" Operator
  .   .   .               UnaryExpr
  .   .   .                 "!" Operator
  .   .   .                   LiteralExpr
  .   .   .                     "false" BooleanLiteral
  .   .   . VarDeclStmt
  .   .   .   VarDecl
  .   .   .     BOOLEAN BaseType
  .   .   .     "b" varname
  .   .   .   BinaryExpr
  .   .   .     "||" Operator
  .   .   .       LiteralExpr
  .   .   .         "false" BooleanLiteral
  .   .   .       BinaryExpr
  .   .   .         "&&" Operator
  .   .   .           BinaryExpr
  .   .   .             "==" Operator
  .   .   .               LiteralExpr
  .   .   .                 "true" BooleanLiteral
  .   .   .               BinaryExpr
  .   .   .                 "<" Operator
  .   .   .                   LiteralExpr
  .   .   .                     "2" IntLiteral
  .   .   .                   BinaryExpr
  .   .   .                     "-" Operator
  .   .   .                       UnaryExpr
  .   .   .                         "-" Operator
  .   .   .                           LiteralExpr
  .   .   .                             "3" IntLiteral
  .   .   .                       BinaryExpr
  .   .   .                         "/" Operator
  .   .   .                           LiteralExpr
  .   .   .                             "4" IntLiteral
  .   .   .                           LiteralExpr
  .   .   .                             "5" IntLiteral
  .   .   .           UnaryExpr
  .   .   .             "!" Operator
  .   .   .               UnaryExpr
  .   .   .                 "!" Operator
  .   .   .                   LiteralExpr
  .   .   .                     "false" BooleanLiteral
=============================================
